rules_go now requires `rulesGoStdlibPrefix` to be set via `-X`

**Describe the bug**

`rules_go` is planning a new release (0.42). This new release will contain the following change: https://github.com/bazelbuild/rules_go/pull/3701.

It leads to this error when using the go toolchain provided by rules_nixpkgs:

```
ERROR: /home/malte/.cache/bazel/_bazel_malte/bb1175a64d2f0f71785aea697a0484ea/external/io_bazel_rules_go/BUILD.bazel:42:7: GoStdlib external/io_bazel_rules_go/stdlib_/pkg failed: (Exit 2): builder failed: error executing command (from target @io_bazel_rules_go//:stdlib) 
  (cd /home/malte/.cache/bazel/_bazel_malte/bb1175a64d2f0f71785aea697a0484ea/sandbox/linux-sandbox/80/execroot/constellation && \
  exec env - \
    CGO_ENABLED=0 \
    GOARCH=amd64 \
    GOEXPERIMENT='' \
    GOOS=linux \
    GOPATH='' \
    GOROOT=external/go_sdk \
    GOROOT_FINAL=GOROOT \
    GOTOOLCHAIN=local \
    PATH=external/zig_sdk/tools:external/zig_sdk/tools/x86_64-linux-gnu.2.23:/bin:/usr/bin \
  bazel-out/k8-opt-exec-C7777A24-ST-e336f88b9a0e/bin/external/go_sdk/builder stdlib -sdk external/go_sdk -installsuffix linux_amd64 -tags netgo,netgo -out bazel-out/k8-opt-ST-04e6e76229c2/bin/external/io_bazel_rules_go/stdlib_ -package std -gcflags '')
# Configuration: 4114e72d42dffca035c92c4853a59204220a10d133544398bcba8b927789b8dc
# Execution platform: @rules_nixpkgs_core//platforms:host

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
panic: rulesGoStdlibPrefix should have been set via -X

goroutine 1 [running]:
main.init.0()
        ./stdliblist.go:113 +0x31
```

**To Reproduce**
Steps to reproduce the behavior.

Use `rules_go` >= https://github.com/bazelbuild/rules_go/commit/c2406b27f77e186d0fb3548d8526e6c14a3236d4
And rules_nixpkgs with `nixpkgs_go_configure`.

**Expected behavior**

Builds should succeed.

**Environment**
 - OS name + version: Ubuntu 22.04
 - Version of the code: 7e627d76ba65d6c42586fc265e46bd370672b4eb

**Additional context**

Happy to provide a fix. As of now, I just don't understand where to take the stdlib prefix from (or if `rules_nixpkgs` even provides this)

--- a/toolchains/go/go.bzl
+++ b/toolchains/go/go.bzl
@@ -211,13 +211,18 @@ nixpkgs_go_toolchain = repository_rule(
 go_sdk_build = """
 load("@{rules_go}//go/private/rules:binary.bzl", "go_tool_binary")
 load("@{rules_go}//go/private/rules:sdk.bzl", "package_list")
-load("@{rules_go}//go:def.bzl", "go_sdk")
+load("@{rules_go}//go:def.bzl", "go_sdk", "RULES_GO_VERSION")
 load("@{helpers}//:go_sdk.bzl", "go_sdk_for_arch")
 load(":go_version.bzl", "go_version")
 
 package(default_visibility = ["//visibility:public"])
 
 go_sdk_for_arch(go_version)
+
+# rules_go >= 0.42.0 requires a stdlib prefix
+TOOL_BINARY_NEEDS_STDLIB_PREFIX = (int(RULES_GO_VERSION.split(".")[0]) > 0) or (int(RULES_GO_VERSION.split(".")[1]) >= 42)
+None if TOOL_BINARY_NEEDS_STDLIB_PREFIX else \
+    print("WARNING: rules_go %s is deprecated and support will soon be removed. Upgrade to rules_go >= 0.42.0. See https://github.com/tweag/rules_nixpkgs/issues/421 for more information." % (RULES_GO_VERSION))
 
 filegroup(
     name = "headers",
@@ -237,6 +242,7 @@ filegroup(
 go_tool_binary(
     name = "builder",
     srcs = ["@{rules_go}//go/tools/builders:builder_srcs"],
+    ldflags = "-X main.rulesGoStdlibPrefix=@{rules_go}//stdlib:" if TOOL_BINARY_NEEDS_STDLIB_PREFIX else None,
     sdk = ":go_sdk",
 )
 

